tutorials/navigation/navigation_using_navigationpathqueryobjects.rst

```diff
@@ -0,0 +1,66 @@
+.. _doc_navigation_using_navigationpathqueryobjects:
+
+Using NavigationPathQueryObjects
+================================
+
+``NavigationPathQueryObjects`` can be used together with ``NavigationServer.query_path()`` 
+to obtain a heavily **customized** navigation path including optional **meta data** about the path.
+
+This requires more setup compared to obtaining a normal NavigationPath but allows to custom-fit 
+the pathfinding and provided path data to the different needs of a project.
+
+NavigationPathQueryObjects consist of a pair of objects, a ``NavigationPathQueryParameters`` object holding the customization options 
+for the query and a ``NavigationPathQueryResult`` that receives (regular) updates with the resulting path and meta data from the query.
+
+2D and 3D versions of ``NavigationPathQueryParameters`` are available as 
+:ref:`NavigationPathQueryParameters2D<class_NavigationPathQueryParameters2D>` and 
+:ref:`NavigationPathQueryParameters3D<class_NavigationPathQueryParameters3D>` respectively.
+
+2D and 3D versions of ``NavigationPathQueryResult`` are available as 
+:ref:`NavigationPathQuerResult2D<class_NavigationPathQueryResult2D>` and 
+:ref:`NavigationPathQueryResult3D<class_NavigationPathQueryResult3D>` respectively.
+
+Both parameters and result are used as a pair with the ``NavigationServer.query_path()`` function.
+
+For the available customization options and their use see the class doc of the parameters.
+
+While not a strict requirement, both objects are intended to be created once in advance, stored in a 
+persistent variable for the agent and reused for every followup path query with updated parameters. 
+This reuse avoids performance implications from frequent object creation if a project 
+has a large quantity of simultaneous agents that regularly update their paths.
+
+.. tabs::
+ .. code-tab:: gdscript GDScript
+
+    extends Node2D
```

Comment:
Both this and the Node3D code examples are giving errors when put into a script in Beta8. The non variable declaration lines of code all have the error `Unexpected "Identifier" in class body.`
![NavError](https://user-images.githubusercontent.com/6080368/207501105-9e1b45a9-7dd4-495c-ae4b-754e8372794e.PNG)

Reply:
That is expected and the case in many other short code snippets that should be copied inside own functions. They are not full functional scripts already wrapped with a random function name so of course the parser complains. The "extends" at the beginning is only there so it is immediately obvious that they are for 2D or 3D.

Reply:
I think it would be better to remove the "extends" from the top of the script and just state before the code block "This is for 2D navigation" or something along those lines, it doesn't have to be that exactly.